bitsandbytes/optim/optimizer.py

```diff
@@ -195,7 +197,8 @@ def cast(param, value):
             elif isinstance(value, dict):
                 for k, v in value.items():
                     if k in self.non_castable_tensor_keys:
-                        value[k] = v.to(param.device)
+                        if move_to_device:
+                            value[k] = v.to(param.device)
```

Comment:
```suggestion
                        buffer = self.get_state_buffer(v, v.dtype)
                        buffer.copy_(v)
                        value[k] = buffer
```

Reply:
Thanks @hansonw! This seems reasonable as a separate PR!